### PR TITLE
Some small fixes

### PR DESCRIPTION
**Pull request by anton-rudeshko**
See comments for details

### CODE REVIEW

File: glisse.js [hunk 1]
```diff
@@ -9,7 +9,7 @@
 * Based on jQuery Plugin Boilerplate 1.3
 *
 */
-(function ($) {
+(function ($, document) {
```
Comment (anton-rudeshko) [inline glisse.js:12]: Using parameter instead of global variable is better for minification, because you using `document` a lot

File: glisse.js [hunk 2]
```diff
@@ -28,6 +28,7 @@
             pictureUrl,
             group,
             isChange = false,
+            mobile = {},
```
Comment (anton-rudeshko) [inline glisse.js:31]: Downwards you define global variable `mobile`. It should be local.

File: glisse.js [hunk 3]
```diff
@@ -37,13 +38,9 @@
         plugin.init = function () {
             plugin.settings = $.extend({}, defaults, options);
 
-            // Mobile ?
-            $.MobileDevice = ((navigator.userAgent.match(/iPhone/i)) || (navigator.userAgent.match(/iPod/i)) || (navigator.userAgent.match(/Android/i)));
-            $.Tablet = ((navigator.userAgent.match(/iPad/i)));
```
Comment (anton-rudeshko) [inline glisse.js:45]: 1. Why pollute jQuery object with custom props?
2. Too much regexps =)

File: glisse.js [hunk 4]
```diff
@@ -77,19 +74,13 @@
                         scrollX: null
                     };
 
-                    document.ontouchmove = function(e){ touchHandler(e); };
-                    document.ontouchstart = function(e){ touchHandler(e); };
-                    document.ontouchend = function(e){ touchHandler(e); };
-
-                    if (!isSupportFixed()) {
```
Comment (anton-rudeshko) [inline glisse.js:87]: `isSupportFixed` is undefined, will throw error

File: glisse.js [hunk 5]
```diff
@@ -77,19 +74,13 @@
                         scrollX: null
                     };
 
-                    document.ontouchmove = function(e){ touchHandler(e); };
-                    document.ontouchstart = function(e){ touchHandler(e); };
-                    document.ontouchend = function(e){ touchHandler(e); };
```
Comment (anton-rudeshko) [inline glisse.js:85]: Don't need function wrappers here

File: glisse.js [hunk 6]
```diff
@@ -77,19 +74,13 @@
...
-                    if (!isSupportFixed()) {
-                        //window.scrollTo(0,0);
-                    }
+                    document.ontouchmove = document.ontouchstart = document.ontouchend = touchHandler;
                 }
             });
         };
 
         var preloadImgs = function preloadImgs(){
-            var current, image_urls = [], i, self = this;
```
Comment (anton-rudeshko) [inline glisse.js:95]: `self` is never used

File: glisse.js [hunk 7]
```diff
@@ -102,7 +93,7 @@
                 current.load(loaded(image_urls[i]));
             }
         };
-
```
Comment (anton-rudeshko) [inline glisse.js:108]: Trailing whitespace (and so on on different lines)

File: glisse.js [hunk 8]
```diff
@@ -159,13 +150,11 @@
                 }
                 else if (docElm.mozRequestFullScreen) {
                     docElm.mozRequestFullScreen();
-                    console.log("ok");
```
Comment (anton-rudeshko) [inline glisse.js:165]: No place for debugging ok's

File: glisse.js [hunk 9]
```diff
@@ -209,10 +198,9 @@
 
         var addImage = function addImage(pic) {
             spinner(true);
-            var url = pic,
```
Comment (anton-rudeshko) [inline glisse.js:215]: `url` is no different from `pic`, inlining it

File: glisse.js [hunk 10]
```diff
@@ -264,9 +252,8 @@
                             .css('display','none');
                     }
 
-                    var url = pictureUrl,
```
Comment (anton-rudeshko) [inline glisse.js:270]: Same as https://github.com/Victa/Glisse.js/pull/11/files#r4100772 here

File: glisse.js [hunk 11]
```diff
@@ -492,7 +478,7 @@
             }
         };
 
-        // === Contols actions  =================
+        // === Controls actions  =================
```
Comment (anton-rudeshko) [inline glisse.js:484]: Typo

File: glisse.js [hunk 12]
```diff
@@ -520,11 +506,7 @@
 
         // Spinner =========================================
         var spinner = function spinner(action) {
-            if(action){
-                plugin.els['overlay'].addClass('loading');
-            } else {
-                plugin.els['overlay'].removeClass('loading');
-            }
+            plugin.els['overlay'].toggleClass('loading', action);
```
Comment (anton-rudeshko) [inline glisse.js:512]: Too much code for toggling class (see http://api.jquery.com/toggleClass/)

File: glisse.js [hunk 13]
```diff
@@ -556,9 +538,7 @@
         })();
 
         var isValidEffect = function isValidEffect(effect){
-            var fx = ['bounce','fadeBig','fade','roll','rotate','flipX','flipY'];
-            if(typeof(effect)=='string' && isNaN(effect) && $.inArray(effect, fx) !== -1)
-                return true;
+            return !!~$.inArray(effect, ['bounce', 'fadeBig', 'fade', 'roll', 'rotate', 'flipX', 'flipY']);
```
Comment (anton-rudeshko) [inline glisse.js:544]: Y so much chekz? ლ(ಠ益ಠლ) 
:smiley: 
`~indexOf` is efficient kinda shortcut for `indexOf > -1`, feel free to check
`!!` to cast to boolean

**Comment by anton-rudeshko** [comment]
Feel free to discuss

File: glisse.js (hunk 1 above)
Comment (Victa) [inline glisse.js:12]: you are right

**Comment by Victa** [comment]
This one is awesome :) thanks for your tips and your fixes.

**Comment by Victa** [comment]
Oops I can't auto merge this request.

**Comment by anton-rudeshko** [comment]
Lemme see

**Comment by anton-rudeshko** [comment]
I was just afraid that you will not accept lots of changes :)
I'll fix issue with merge, amend this commit and ping you when it's done

**Comment by anton-rudeshko** [comment]
I've rebased my changes, please try applying now

**Comment by Victa** [comment]
Don't worry. I'm so glad to have pull request on my projects! I appreciate it!

**Pull request merged**